drivers/performance_measurement/app/models/performance_measurement/result_calculation.rb

```diff
@@ -202,115 +172,131 @@ def client_data(field, period, project_id: nil)
         title: detail_title_for(__method__.to_sym),
-        passed: passed?(field, reporting_count, comparison_count),
         direction: direction(field, reporting_count, comparison_count),
         primary_value: reporting_count,
         primary_unit: 'clients',
-        secondary_value: percent_changed(reporting_count, comparison_count),
+        secondary_value: progress,
         secondary_unit: '%',
         value_label: 'Change over year',
         comparison_primary_value: comparison_count,
         system_level: project&.project_id.blank?,
         project_id: project&.project_id,
-        goal: goal(field),
+        passed: progress[:passed],
+        goal: progress[:goal],
+        goal_progress: progress[:progress],
       )
     end
 
-    def first_time_homeless_clients(field, project: nil)
+    def first_time_homeless_clients(detail, project: nil)
+      field = detail[:calculation_column]
       reporting_count = client_count(field, :reporting, project_id: project&.project_id)
       comparison_count = client_count(field, :comparison, project_id: project&.project_id)
 
+      progress = calculate_processed(detail[:goal_calculation], reporting_count, comparison_count)
       PerformanceMeasurement::Result.new(
         report_id: id,
         field: __method__,
         title: detail_title_for(__method__.to_sym),
-        passed: passed?(field, reporting_count, comparison_count),
         direction: direction(field, reporting_count, comparison_count),
         primary_value: reporting_count,
         primary_unit: 'clients',
-        secondary_value: percent_changed(reporting_count, comparison_count),
+        secondary_value: progress,
         secondary_unit: '%',
         value_label: 'Change over year',
         comparison_primary_value: comparison_count,
         system_level: project&.project_id.blank?,
         project_id: project&.project_id,
-        goal: goal(field),
+        passed: progress[:passed],
+        goal: progress[:goal],
+        goal_progress: progress[:progress],
       )
     end
 
-    def length_of_homeless_time_homeless_average(field, project: nil)
+    def length_of_homeless_time_homeless_average(detail, project: nil)
       return unless project.blank? || project.hud_project&.es? || project.hud_project&.sh? || project.hud_project&.th?
 
+      field = detail[:calculation_column]
+
       reporting_count = client_count_present(field, :reporting, project_id: project&.project_id)
       comparison_count = client_count_present(field, :comparison, project_id: project&.project_id)
       reporting_days = client_sum(field, :reporting, project_id: project&.project_id)
       comparison_days = client_sum(field, :comparison, project_id: project&.project_id)
       reporting_average = average(reporting_days, reporting_count)
       comparison_average = average(comparison_days, comparison_count)
 
+      progress = calculate_processed(detail[:goal_calculation], reporting_average, nil)
```

Comment:
I would use default instead of specifying `nil`, or remove the default to make it explicit.